crates/fluvio-sc/src/stores/topic/store.rs

```diff
@@ -19,42 +17,28 @@ pub type DefaultTopicLocalStore = TopicLocalStore<u32>;
 
 #[async_trait]
 pub trait TopicMd<C: MetadataItem> {
-    async fn create_new_partitions(
-        &self,
-        partition_store: &PartitionLocalStore<C>,
-    ) -> Vec<PartitionMetadata<C>>;
+    async fn partitions_from_replicas(&self) -> Vec<PartitionMetadata<C>>;
 }
 
 #[async_trait]
 impl<C: MetadataItem> TopicMd<C> for TopicMetadata<C>
 where
     C: MetadataItem + Send + Sync,
 {
-    /// create new partitions from my replica map if it doesn't exists
-    async fn create_new_partitions(
-        &self,
-        partition_store: &PartitionLocalStore<C>,
-    ) -> Vec<PartitionMetadata<C>> {
-        let mut partitions = vec![];
-        let replica_map = &self.status.replica_map;
-        trace!(?replica_map, "creating new partitions for topic");
-        for (idx, replicas) in replica_map.iter() {
-            let mirror = self.status.mirror_map.get(idx);
-
-            let replica_key = ReplicaKey::new(self.key(), *idx);
-
-            let partition_spec = PartitionSpec::from_replicas(replicas.clone(), &self.spec, mirror);
-            if !partition_store.contains_key(&replica_key).await {
-                debug!(?replica_key, ?partition_spec, "creating new partition");
-                partitions.push(
-                    MetadataStoreObject::with_spec(replica_key, partition_spec)
-                        .with_context(self.ctx.create_child()),
-                )
-            } else {
-                debug!(?replica_key, "partition already exists");
-            }
-        }
-        partitions
+    /// get partitions from replica map
+    async fn partitions_from_replicas(&self) -> Vec<PartitionMetadata<C>> {
```

Comment:
this need unit test

Reply:
Sure! I added one test for `partitions_from_replicas`